services/gateway_manager/src/main/java/com/futurewei/alcor/gatewaymanager/dao/GWAttachmentRepository.java

```diff
@@ -0,0 +1,72 @@
+package com.futurewei.alcor.gatewaymanager.dao;
+
+import com.futurewei.alcor.common.db.CacheException;
+import com.futurewei.alcor.common.db.CacheFactory;
+import com.futurewei.alcor.common.db.ICache;
+import com.futurewei.alcor.common.db.Transaction;
+import com.futurewei.alcor.common.db.repo.ICacheRepository;
+import com.futurewei.alcor.gatewaymanager.entity.GWAttachment;
+import com.futurewei.alcor.web.entity.gateway.GatewayEntity;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Repository
+public class GWAttachmentRepository implements ICacheRepository<GWAttachment> {
+
```

Comment:
@songxiaoyan do we need cache for GWAttachment?

Reply:
I think we need to create the Cache Cache of GatewayEntity and GwAttachment in GM's DB for later interface query

Reply:
Ok, agree.